src/WidgetBase.ts

```diff
@@ -193,6 +193,8 @@ export class WidgetBase<P = WidgetProperties, C extends DNode = DNode> extends E
 
 	private _numRootNodes = 0;
 
+	private _rootNodeKeys: object[];
```

Comment:
is a root node key really an object?

Reply:
Got a typing error when i set it to `string[]`. The typedoc popup said that `properties.key` is an object.
![screenshot 2017-09-15 09 44 15](https://user-images.githubusercontent.com/814453/30474133-8958ada2-99fa-11e7-8404-4062e68a7238.png)